File: base/src/main/java/io/quarkus/code/rest/exceptions/CodestartExceptionMapper.java
package io.quarkus.code.rest.exceptions;

import io.quarkus.devtools.codestarts.CodestartException;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;

@Provider
public class CodestartExceptionMapper implements ExceptionMapper<CodestartException> {
    @Override
    public Response toResponse(CodestartException e) {
        String message = "Bad request > " + e.getMessage();
        return Response.status(Response.Status.BAD_REQUEST)
                .entity(message)
                .type(MediaType.TEXT_PLAIN)
                .build();
    }
}

File: base/src/main/kotlin/io/quarkus/code/rest/exceptions/CodestartExceptionMapper.kt
package io.quarkus.code.rest.exceptions

import io.quarkus.devtools.codestarts.CodestartException
import jakarta.ws.rs.core.MediaType.TEXT_PLAIN
import jakarta.ws.rs.core.Response
import jakarta.ws.rs.ext.ExceptionMapper
import jakarta.ws.rs.ext.Provider

@Provider
class CodestartExceptionMapper: ExceptionMapper<CodestartException>  {
    override fun toResponse(e: CodestartException): Response {
        val message = "Bad request > ${e.message}"
        return Response.status(Response.Status.BAD_REQUEST)
            .entity(message)
            .type(TEXT_PLAIN)
            .build()
    }
}